devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy

```diff
@@ -316,8 +321,8 @@ def updated(){
             sendEvent(name: 'configuration',value: 0, descriptionText: "Settings changed and will update at next report. Measure interval set to ${interval} mins")
     	}
     }
-    //check every 1 hour + 2mins
-    sendEvent(name: "checkInterval", value: 1 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    //check every 3 hours
+    sendEvent(name: "checkInterval", value: 3 * 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```

Comment:
It looks like attribute 0x0000 on cluster 0x0405 is configured for reporting every 610 seconds ~ 10 minutes. Why the increase of checkInterval to 3 hours?

Reply:
Yes, the default interval is 610 s, however, in the preferences the user can change the interval to 2 hours, so I had made it 3 to cover the longest interval.  If its preferable to keep it shorter, 2 hours would work, but would prefer to make it a little longer than the previous 1 hour, if possible.
line 342:
> //update intervals
def intervalUpdate(){
	log.debug "intervalUpdate"
    def minReport = 10
    def maxReport = 610
    if (interval != null) {
    	minReport = interval
        maxReport = interval * 61
    }
    [    
    	"zcl global send-me-a-report 0x405 0x0000 0x21 $minReport $maxReport {6400}", "delay 500",
    	"send 0x${device.deviceNetworkId} 1 1", "delay 500",
        "zcl global send-me-a-report 1 0x0000 0x21 0x0C 0 {0500}", "delay 500",
        "send 0x${device.deviceNetworkId} 1 1", "delay 500",
    ]    
}

Reply:
You could programmatically change the checkInterval based on that preference. I know we do that in `aeon-multisensor-6`. For longer `checkIntervals` we prefer `reportingInterval * 2 + 10 * 60`.

Reply:
Sounds good, changed install value back to previous 62 min value, and added variable interval for when user changes poll interval.  However, did make minimum value 62 mins.  B/c some of the previous model sensors, would change the "measure" interval but would still only report after an hour.